biomage/utils/AuroraClient.py

```diff
@@ -134,13 +134,11 @@ def __exit__(self, exc_type, exc_value, tb):
         self.close_tunnel()
 
     def open_tunnel(self):
-        if self.local_port is not None:
-            return
-
-        free_port = _find_free_port()
-        if free_port is None:
-            raise Exception("No free port between 5432 and 6000")
-        self.local_port = free_port
+        if self.local_port is None:
```

Comment:
could you explain this change?

Reply:
This change opens a tunnel when a port is specified and we use context manager (`with AuroraClient() ...`).

Before this change, when we want to use a specific port with `AuroraClient`, we have to open the tunnel first then use the port with the `AuroraClient`. I think this is redundant because AuroraClient has methods to manage port opening and closing. This change allows a user to specify a port that we want to use with the `AuroraClient`, and the client will manage the port (open, then close when done), if we use context manager.

Specifying what port to use with AuroraClient is important in the case of migrations because right now the port that is used in the knexfile connections are hard coded. 5431 for local migrations, and 5432 for non-local migrations.

Reply:
oh I see I didn't see the indentation after this code, thanks